Spigot-Server-Patches/0269-Implement-extended-PaperServerListPingEvent.patch

```diff
@@ -0,0 +1,253 @@
+             this.q.setPlayerSample(new ServerPing.ServerPingPlayerSample(this.I(), this.H()));
+-            GameProfile[] agameprofile = new GameProfile[Math.min(this.H(), 12)];
++            GameProfile[] agameprofile = new GameProfile[Math.min(this.H(), org.spigotmc.SpigotConfig.playerSample)]; // Paper
+             int j = MathHelper.nextInt(this.r, 0, this.H() - agameprofile.length);
+ 
+             for (int k = 0; k < agameprofile.length; ++k) {
+@@ -1116,10 +1116,12 @@ public abstract class MinecraftServer implements ICommandListener, Runnable, IAs
+         return "1.12.2";
+     }
+ 
++    public int getPlayerCount() { return H(); } // Paper - OBFHELPER
+     public int H() {
+         return this.v.getPlayerCount();
+     }
+ 
++    public int getMaxPlayers() { return I(); } // Paper - OBFHELPER
+     public int I() {
+         return this.v.getMaxPlayers();
+     }
+diff --git a/src/main/java/net/minecraft/server/PacketStatusListener.java b/src/main/java/net/minecraft/server/PacketStatusListener.java
+index 313bb0007..f3c25367d 100644
+--- a/src/main/java/net/minecraft/server/PacketStatusListener.java
++++ b/src/main/java/net/minecraft/server/PacketStatusListener.java
+@@ -30,6 +30,8 @@ public class PacketStatusListener implements PacketStatusInListener {
+             this.networkManager.close(PacketStatusListener.a);
+         } else {
+             this.d = true;
++            // Paper start - Replace everything
++            /*
+             // CraftBukkit start
+             // this.networkManager.sendPacket(new PacketStatusOutServerInfo(this.minecraftServer.getServerPing()));
+             final Object[] players = minecraftServer.getPlayerList().players.toArray();
+@@ -125,6 +127,9 @@ public class PacketStatusListener implements PacketStatusInListener {
+             ping.setServerInfo(new ServerPing.ServerData(minecraftServer.getServerModName() + " " + minecraftServer.getVersion(), version));
+ 
+             this.networkManager.sendPacket(new PacketStatusOutServerInfo(ping));
++            */
++            com.destroystokyo.paper.network.StandardPaperServerListPingEventImpl.processRequest(this.minecraftServer, this.networkManager);
++            // Paper end
+         }
+         // CraftBukkit end
+     }
+diff --git a/src/main/java/net/minecraft/server/ServerPing.java b/src/main/java/net/minecraft/server/ServerPing.java
+index 10bc7fc1f..f27563875 100644
+--- a/src/main/java/net/minecraft/server/ServerPing.java
++++ b/src/main/java/net/minecraft/server/ServerPing.java
+@@ -29,6 +29,8 @@ public class ServerPing {
+         this.a = ichatbasecomponent;
+     }
+ 
++    public ServerPingPlayerSample getPlayers() { return b(); } // Paper - OBFHELPER
++
+     public ServerPing.ServerPingPlayerSample b() {
+         return this.b;
+     }
+@@ -152,6 +154,9 @@ public class ServerPing {
+             this.b = j;
+         }
+ 
++        public GameProfile[] getSample() { return c(); } // Paper - OBFHELPER
```

Comment:
OBFHELPERs should be ideally placed above what they're calling, that way they conflict on merge instead of later on in the process

Reply:
And getters/setters ideally just place on same line as the field so it's a guaranteed conflict (wiggle can be too smart and byte us on above the methods)